src/Components/test/testassets/TestServer/Pages/Transports.cshtml

```diff
@@ -0,0 +1,66 @@
+﻿@page
+@addTagHelper "*, Microsoft.AspNetCore.Mvc.TagHelpers"
+
+<root><component type="typeof(BasicTestApp.Index)" render-mode="Server" /></root>
+
+<div id="blazor-error-ui">
+    An unhandled exception has occurred. See browser dev tools for details.
+    <a href="" class="reload">Reload</a>
+    <a class="dismiss">🗙</a>
+</div>
+
+<button id="startBlazorServerBtn" onclick="startBlazorServer()">Start Normally</button>
+<button id="startWithWebSocketsDisabledInBrowserBtn" onclick="startWithWebSocketsDisabledInBrowser()">Start with WebSockets Disabled in Browser</button>
+<button id="startWithLongPollingBtn" onclick="startWithLongPolling()">Start with Long Polling</button>
+<button id="startAndRejectWebSocketConnectionBtn" onclick="startAndRejectWebSocketConnection()">Start with WebSockets and Reject Connection</button>
+
+<script src="_framework/blazor.server.js" autostart="false"></script>
+<script>
+    console.log('Blazor server-side');
+
+    function startBlazorServer() {
+        Blazor.start({
+            logLevel: 1, // LogLevel.Debug
+            configureSignalR: builder => {
+                builder.configureLogging("debug") // LogLevel.Debug
+            }
+        }).then(function () {
+            window['__aspnetcore__testing__blazor__start__script__executed__'] = true;
+        });
+    }
+
+    function startWithWebSocketsDisabledInBrowser() {
+        WebSocket = null; // emulates browsers which don't support WebSockets
+        startBlazorServer();
+    }
+
+    function startWithLongPolling() {
+        Blazor.start({
+            logLevel: 1, // LogLevel.Debug
+            configureSignalR: builder => {
+                builder.configureLogging("debug") // LogLevel.Debug
+                    .withUrl('_blazor', 4) // Long Polling (4)
+            }
+        }).then(function () {
+            window['__aspnetcore__testing__blazor__start__script__executed__'] = true;
+        });
+    }
+
+    function WebSocketNotAllowed() { throw new Error("Don't allow Websockets."); }
+
+    function startAndRejectWebSocketConnection() {
+        Blazor.start({
+            logLevel: 1, // LogLevel.Debug
+            configureSignalR: builder => {
+                builder.configureLogging("debug") // LogLevel.Debug
+                        .withUrl('_blazor',
+                        {
+                            WebSocket: WebSocketNotAllowed,
```

Comment:
Same for this one. A much more useful test is to have the server disable websockets which would simulates a misconfigured server.

Reply:
I'm inclined to keep this test as:

> WebSocketNotAllowed approach should effectively be the same as the server rejecting a websocket request

It's part of SignalR's [testing](https://github.com/dotnet/aspnetcore/blob/fd85422b45f6c080b440648656807945e418cde6/src/SignalR/clients/ts/signalr/tests/HttpConnection.test.ts#L252) [mechanism](https://github.com/dotnet/aspnetcore/blob/fd85422b45f6c080b440648656807945e418cde6/src/SignalR/clients/ts/signalr/src/IHttpConnectionOptions.ts#L47-L52) that we're leveraging.